.github/workflows/pytest-legacy.yml

```diff
@@ -22,13 +22,42 @@ jobs:
       with:
         python-version: '3.10'
 
-    - name: Install specific out-dated version of dependencies
+    #----------------------------------------------
+    #  -----  install & configure poetry  -----
+    #----------------------------------------------
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
+      with:
+        virtualenvs-create: true
+        virtualenvs-in-project: true
+        installer-parallel: true
+    #----------------------------------------------
+    #       load cached venv if cache exists
+    #----------------------------------------------
+    - name: Load cached venv
+      id: cached-poetry-dependencies
+      uses: actions/cache@v4
+      with:
+        path: .venv
+        key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
```

Comment:
That depends on when the venv is cached, I think. If I understand correctly, but I can't quickly find some docs about this to check, the venv is probably cached after/by the `poetry install` command based on the `poetry.lock` file. So as long as the lock file on `main` stays unchanged, I'd think the pinned dependencies are added once, a new cache is generated, and the next time this workflow runs, this cache can be used. We still add the pinned versions, but `poetry install` should be smart enough to recognize everything it needs is already present.

Reply:
But this is a good point, the caches we now have don't register the names correctly. So at the moment, they are called like this: venv-Windows--10eb3670e2cac9f6987670ba90f2dc830dfc72ed64d8fab506017c5c6f03c048, when it should be something like venv-Windows-3.12-10eb3670e2cac9f6987670ba90f2dc830dfc72ed64d8fab506017c5c6f03c048. I'll push a fix for just the pytest-legacy workflow, maybe this will help show if the cache is being used as I expect it to be.

Reply:
If you go to https://github.com/IAMconsortium/pyam/actions/caches now, you will see that "venv-Linux-3.10.13-2c3debcd9ff0663e5777cf119982f37c8e4d5459d475573c952c825a0057874a", the oldest cache, so the one from pytest-legacy, is being used again and again in newer workflows. 
Now, I'm still not sure if this means everything is working as expected, but to me, it almost seems fine because if the same poetry.lock can be used for subsequent loading, at least parts of the cache must be reusable, right?
Maybe we could make this more explicit by running `poetry lock` after all `poetry add`s?